src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -233,23 +233,33 @@ def __call__(
         """
         return super().__call__(inputs, **kwargs)
 
-    def _sanitize_parameters(self, **kwargs):
+    def _sanitize_parameters(
+        self,
+        chunk_length_s=None,
+        stride_length_s=None,
+        ignore_warning=None,
+        decoder_kwargs=None,
+        return_timestamps=None,
+        **generate_kwargs
+    ):
         # No parameters on this pipeline right now
         preprocess_params = {}
-        if "chunk_length_s" in kwargs:
-            preprocess_params["chunk_length_s"] = kwargs["chunk_length_s"]
-        if "stride_length_s" in kwargs:
-            preprocess_params["stride_length_s"] = kwargs["stride_length_s"]
-        if "ignore_warning" in kwargs:
-            preprocess_params["ignore_warning"] = kwargs["ignore_warning"]
+        if chunk_length_s is not None:
+            preprocess_params["chunk_length_s"] = chunk_length_s
+        if stride_length_s is not None:
+            preprocess_params["stride_length_s"] = stride_length_s
+        if ignore_warning is not None:
+            preprocess_params["ignore_warning"] = ignore_warning
+
+        forward_params = generate_kwargs
```

Comment:
```suggestion
        if max_new_tokens is not None:
            forward_params["generate_kwargs"] = {}
            forward_params['generate_kwargs"]["max_new_tokens"] = max_new_tokens
        if generate_kwargs is not None:
            if max_new_tokens is not None:
                raise ValueError("`max_new_tokens` is defined both as an argument and inside `generate_kwargs` argument, please use only 1 version")
            forward_params["generate_kwargs"].update(generate_kwargs)
```

Reply:
It won't pass `if max_new_tokens is None and generate_kwargs is not None`. What about this way @Narsil ?

```python
forward_params = {"generate_kwargs": {}}
if max_new_tokens is not None:
    forward_params['generate_kwargs"]["max_new_tokens"] = max_new_tokens
if generate_kwargs is not None:
    if max_new_tokens is not None and "max_new_tokens" in generate_kwargs:
        raise ValueError("`max_new_tokens` is defined both as an argument and inside `generate_kwargs` argument, please use only 1 version")
    forward_params["generate_kwargs"].update(generate_kwargs)
```

Reply:
Much better, it's adctually what I had in mind ! :)